model/DataObject.php

```diff
@@ -1916,25 +1912,31 @@ public function db($fieldName = null) {
 			$classes = array_reverse($classes);
 		}
 
-		$items = array();
+		$db = array();
 		foreach($classes as $class) {
-			if(isset(self::$_cache_db[$class])) {
-				$dbItems = self::$_cache_db[$class];
-			} else {
-				$dbItems = (array) Config::inst()->get($class, 'db', Config::UNINHERITED);
-				self::$_cache_db[$class] = $dbItems;
-			}
-
-			if($fieldName) {
-				if(isset($dbItems[$fieldName])) {
-					return $dbItems[$fieldName];
+			// Merge fields with new fields and composite fields
+			$fields = self::database_fields($class);
+			$compositeFields = self::composite_fields($class, false);
+			$db = array_merge($db, $fields, $compositeFields);
+
+			// Check for search field
+			if($fieldName && isset($db[$fieldName])) {
+				// Set table for the given field
+				if($fieldName === 'ID') {
```

Comment:
It doesn't matter; Fixed fields will already be assigned to the base table without this code. The reason for ID being an exception here is because it is the only field that is on EVERY table, so it should default to the best (base) table.

However, I'll change this code to make it more explicitly map fixed_fields to the base. It won't hurt and hopefully will improve readability. :D